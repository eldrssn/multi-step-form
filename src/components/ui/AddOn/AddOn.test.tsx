import { render } from '@testing-library/react';
import { AddOn } from './';

describe('AddOn', () => {
  test('if isChecked is true', () => {
    const { getByTestId, getByRole, container } = render(
      <AddOn
        addOn={{
          title: 'Title',
          description: 'Description',
          priceYearly: '100',
          priceMonthly: '10',
        }}
        onChange={jest.fn()}
        isChecked={true}
        isYearly={true}
      />
    );

    const label = getByTestId('addon');
    expect(label).toHaveClass('checked');

    const checkboxInput = getByRole('checkbox');
    expect(checkboxInput).toHaveProperty('checked', true);
    expect(container).toMatchSnapshot();
  });

  test('if isChecked is false', () => {
    const { getByTestId, getByRole } = render(
      <AddOn
        addOn={{
          title: 'Title',
          description: 'Description',
          priceYearly: '100',
          priceMonthly: '10',
        }}
        onChange={jest.fn()}
        isChecked={false}
        isYearly={true}
      />
    );

    const label = getByTestId('addon');
    expect(label).not.toHaveClass('checked');

    const checkboxInput = getByRole('checkbox');
    expect(checkboxInput).toHaveProperty('checked', false);
  });

  test('displays correct price based on isYearly is true', () => {
    const { getByText } = render(
      <AddOn
        addOn={{
          title: 'Title',
          description: 'Description',
          priceYearly: '100',
          priceMonthly: '10',
        }}
        onChange={jest.fn()}
        isChecked={false}
        isYearly={true}
      />
    );

    const yearlyPrice = getByText('+$100/yr');
    expect(yearlyPrice).toBeInTheDocument();
  });

  test('displays correct price based on isYearly is false', () => {
    const { getByText } = render(
      <AddOn
        addOn={{
          title: 'Title',
          description: 'Description',
          priceYearly: '100',
          priceMonthly: '10',
        }}
        onChange={jest.fn()}
        isChecked={false}
        isYearly={false}
      />
    );

    const monthlyPrice = getByText('+$10/mo');
    expect(monthlyPrice).toBeInTheDocument();
  });

  test('calls onChange when the checkbox is clicked', () => {
    const onChangeMock = jest.fn();
    const { getByRole } = render(
      <AddOn
        addOn={{
          title: 'Title',
          description: 'Description',
          priceYearly: '100',
          priceMonthly: '10',
        }}
        onChange={onChangeMock}
        isChecked={false}
        isYearly={true}
      />
    );

    const checkboxInput = getByRole('checkbox');
    checkboxInput.click();

    expect(onChangeMock).toHaveBeenCalled();
  });

  test('displays correct text data', () => {
    const { getByText } = render(
      <AddOn
        addOn={{
          title: 'Title',
          description: 'Description',
          priceYearly: '100',
          priceMonthly: '10',
        }}
        onChange={jest.fn()}
        isChecked={false}
        isYearly={true}
      />
    );

    expect(getByText('Title')).toBeInTheDocument();
    expect(getByText('Description')).toBeInTheDocument();
  });
});
