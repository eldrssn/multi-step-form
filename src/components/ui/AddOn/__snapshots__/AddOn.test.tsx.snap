// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddOn if isChecked is true 1`] = `
<div>
  <label
    class="addon checked"
    data-testid="addon"
  >
    <input
      checked=""
      type="checkbox"
    />
    <span
      class="checkmark"
    />
    <div
      class="info"
    >
      <p
        class="title"
      >
        Title
      </p>
      <p
        class="description"
      >
        Description
      </p>
    </div>
    <p
      class="price"
    >
      +$
      100
      /
      yr
    </p>
  </label>
</div>
`;
